src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -172,17 +182,41 @@ private function evaluateBracket(string $expr, mixed $value): array
             return $result;
         }
 
-        if (preg_match('/^(-?\d*+)\s*+:\s*+(-?\d*+)(?:\s*+:\s*+(-?\d++))?$/', $expr, $matches)) {
+        if (preg_match('/^(-?\d*+)\s*+:\s*+(-?\d*+)(?:\s*+:\s*+(-?\d*+))?$/', $expr, $matches)) {
             if (!array_is_list($value)) {
                 return [];
             }
 
+            $startStr = trim($matches[1]);
+            $endStr = trim($matches[2]);
+            $stepStr = trim($matches[3] ?? '1');
+
+            if (
+                JsonPathUtils::hasLeadingZero($startStr)
+                || JsonPathUtils::hasLeadingZero($endStr)
+                || JsonPathUtils::hasLeadingZero($stepStr)
+            ) {
+                throw new JsonCrawlerException($expr, 'slice selector numbers cannot have leading zeros');
+            }
+
+            if ('-0' === $startStr || '-0' === $endStr || '-0' === $stepStr) {
+                throw new JsonCrawlerException($expr, 'slice selector cannot contain negative zero');
+            }
+
+            if (
+                JsonPathUtils::isIntegerOverflow($startStr)
+                || JsonPathUtils::isIntegerOverflow($endStr)
+                || JsonPathUtils::isIntegerOverflow($stepStr)
+            ) {
+                throw new JsonCrawlerException($expr, 'slice selector integer overflow');
+            }
+
             $length = \count($value);
-            $start = '' !== $matches[1] ? (int) $matches[1] : null;
-            $end = '' !== $matches[2] ? (int) $matches[2] : null;
-            $step = isset($matches[3]) && '' !== $matches[3] ? (int) $matches[3] : 1;
+            $start = '' !== $startStr ? (int) $startStr : null;
```

Comment:
How does this work if `$startStr` is the representation of an integer that exceeds `PHP_INT_MAX`? Or is lower than `PHP_INT_MINX`?

Reply:
Fixed in the last commit, CI is now green on x86 a well